examples_tests/49.NewAPITest/main.cpp

```diff
@@ -0,0 +1,330 @@
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#define _NBL_STATIC_LIB_
+#include <iostream>
+#include <cstdio>
+#include <nabla.h>
+#include <nbl/system/CWindowWin32.h>
+
+using namespace nbl;
+using namespace core;
+
+int main()
+{
+	constexpr uint32_t WIN_W = 800u;
+	constexpr uint32_t WIN_H = 600u;
+	constexpr uint32_t SC_IMG_COUNT = 3u;
+
+	const char* vs_source = R"(#version 430
+
+layout (location = 0) in vec2 Pos;
+layout (location = 1) in vec3 Color;
+
+layout (location = 0) out vec3 OutColor;
+
+void main()
+{
+	OutColor = Color;
+	gl_Position = vec4(Pos, 0.0, 1.0);
+}
+)";
+	const char* fs_source = R"(#version 430
+
+layout (location = 0) in vec3 InColor;
+layout (location = 0) out vec4 OutColor;
+
+void main()
+{
+	OutColor = vec4(InColor, 1.0);
+}
+)";
+
+	auto win = system::CWindowWin32::create(WIN_W, WIN_H, system::IWindow::ECF_NONE);
+
+	auto gl = video::IAPIConnection::create(video::EAT_OPENGL, 0, "New API Test");
+	auto surface = gl->createSurface(win.get());
+
+	auto gpus = gl->getPhysicalDevices();
+	assert(!gpus.empty());
+	auto gpu = gpus.begin()[0];
+
+	assert(surface->isSupported(gpu.get(), 0u));
+
+	video::ILogicalDevice::SCreationParams dev_params;
+	dev_params.queueParamsCount = 1u;
+	video::ILogicalDevice::SQueueCreationParams q_params;
+	q_params.familyIndex = 0u;
+	q_params.count = 1u;
+	q_params.flags = static_cast<video::IGPUQueue::E_CREATE_FLAGS>(0);
+	float priority = 1.f;
+	q_params.priorities = &priority;
+	dev_params.queueCreateInfos = &q_params;
+	auto device = gpu->createLogicalDevice(dev_params);
+
+	auto* queue = device->getQueue(0u, 0u);
+
+	core::smart_refctd_ptr<video::ISwapchain> sc;
+	{
+		video::ISwapchain::SCreationParams sc_params;
+		sc_params.width = WIN_W;
+		sc_params.height = WIN_H;
+		sc_params.arrayLayers = 1u;
+		sc_params.minImageCount = SC_IMG_COUNT;
+		sc_params.presentMode = video::ISurface::EPM_FIFO_RELAXED;
```

Comment:
which modes will you plan on supporting in GL?

Reply:
idk, ive implemented something basic to have working backend, with more specfic stuff i'll probably need help

Reply:
#145